packages/light-client-experimental/src/smoldot/resources/client.ts

```diff
@@ -0,0 +1,31 @@
+import * as smoldot from "smoldot"
+import * as bindings from "../bindings"
+import { Effect, pipe as $, Runtime } from "effect"
+
+export type AcquireOptions = Omit<smoldot.ClientOptions, "logCallback"> & {
+  logCallback: (
+    ...args: Parameters<NonNullable<smoldot.ClientOptions["logCallback"]>>
+  ) => Effect.Effect<void, never, never>
+}
+
+export const acquire = (
+  options: AcquireOptions,
+): Effect.Effect<bindings.client.Client, never, never> =>
+  $(
+    Effect.runtime<never>(),
+    Effect.andThen(Runtime.runSync),
+    Effect.andThen((runSync) =>
+      Effect.sync(() =>
+        smoldot.start({
+          ...options,
+          logCallback: (...args) => options.logCallback(...args).pipe(runSync),
+        }),
+      ),
+    ),
+    Effect.map(bindings.client.make),
+  )
```

Comment:
I wonder what the standard version of this would look like, because it can be a bit hard following when there are no intermediate variable names etc. I geuss something like this but not really sure:

```ts
const runtime = Effect.runtime<never>();
const res = runtime.runSync(() => {
    smoldot.start({ 
        ...options.
        logCallback: (...args) => Effect.runSync(options.logCallback(...args))
});
bindings.client.make(res)
```

In general I wonder whether we can reduce the amount of `Effect` related code and use it to tie up things where it provides a bunch of value while writing the "lower level" bits in more plain JS so that it's more readable; it's important that others can understand and review these things :)